src/TopoMojo.Hypervisor/Proxmox/ProxmoxClient.cs

```diff
@@ -522,40 +496,44 @@ public async Task<Vm> Save(string id)
             return vm;
         }
 
-        private async Task CompleteSave(Result task, string oldId, int nextId, IClusterResourceVm template, string vmId)
+        private async Task CompleteSave(Result task, string oldId, int nextId, Vm template, string vmId)
         {
             try
             {
-                await this.WaitForTaskToFinish(task);
+                await _pveClient.WaitForTaskToFinish(task);
 
                 if (!task.IsSuccessStatusCode)
                     throw new Exception($"Clone failed: {task.ReasonPhrase}");
 
                 // Convert to template
-                task = await _pveClient.Nodes[template.Node].Qemu[nextId].Template.Template();
-                await this.WaitForTaskToFinish(task);
+                task = await _pveClient.Nodes[template.Host].Qemu[nextId].Template.Template();
+                await _pveClient.WaitForTaskToFinish(task);
 
                 if (!task.IsSuccessStatusCode)
+                {
+                    var destroyTask = await _pveClient.Nodes[template.Host].Qemu[nextId].DestroyVm();
+                    await _pveClient.WaitForTaskToFinish(destroyTask);
                     throw new Exception($"Convert to template failed: {task.ReasonPhrase}");
+                }
 
                 // Delete old vm
                 await this.Delete(oldId);
 
                 // Tag old template
                 // Janitor will delete anything with this tag if deletion fails now
                 task = await _pveClient
-                    .Nodes[template.Node]
-                    .Qemu[template.VmId]
+                    .Nodes[template.Host]
+                    .Qemu[template.Id]
```

Comment:
Same kind of deal here - my goal would be to push all updates to VMs through `PushVmConfigUpdate`. Right now, that function can only update NetAssignments, but I can easily make it do tags too. Let me know!